File: sol/solution/0001-0100/0024. Swap Nodes in Pairs/Solution.kt
/**
 * Example:
 * var li = ListNode(5)
 * var v = li.`val`
 * Definition for singly-linked list.
 * class ListNode(var `val`: Int) {
 *     var next: ListNode? = null
 * }
 */
class Solution {
    fun swapPairs(head: ListNode?): ListNode? {
        if (head?.next == null) return head
        val dummy = ListNode(0)
        dummy.next = head
        var prev: ListNode? = dummy
        while (prev?.next != null && prev.next?.next != null) {
            val first = prev.next
            val second = first?.next
            first?.next = second?.next
            second?.next = first
            prev.next = second
            prev = first
        }
        return dummy.next
    }
}

File: sol/solution/0001-0100/0024. Swap Nodes in Pairs/Solution.swift
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public var val: Int
 *     public var next: ListNode?
 *     public init() { self.val = 0; self.next = nil; }
 *     public init(_ val: Int) { self.val = val; self.next = nil; }
 *     public init(_ val: Int, _ next: ListNode?) { self.val = val; self.next = next; }
 * }
 */
class Solution {
    func swapPairs(_ head: ListNode?) -> ListNode? {
        if head == nil || head?.next == nil {
            return head
        }
        let dummy = ListNode(0, head)
        var prev = dummy
        while prev.next != nil && prev.next?.next != nil {
            let first = prev.next
            let second = first?.next
            first?.next = second?.next
            second?.next = first
            prev.next = second
            prev = first!
        }
        return dummy.next
    }
}